src/model/mod.rs

```diff
@@ -460,8 +460,8 @@ pub struct PairPileup<'a, A, B, P> where
     case_sample_model: likelihood::LatentVariableModel,
     control_sample_model: likelihood::LatentVariableModel,
     // these two caches are useful for PairModels where case and control are independent of each other and Events are just combinations of across the two axes, as e.g. in the single-cell-bulk model
-    case_likelihood_cache: &'a mut BTreeMap<AlleleFreq, LogProb>,
-    control_likelihood_cache: &'a mut BTreeMap<AlleleFreq, LogProb>,
+    case_likelihood_cache: Cell<BTreeMap<AlleleFreq, LogProb>>,
+    control_likelihood_cache: Cell<BTreeMap<AlleleFreq, LogProb>>,
```

Comment:
Why the cell?

Reply:
Oh, someone is watching... ;)

The idea was to keep the PairPileup in general immutable, and only make the fields mutable that need to be. A side product of something I tried when just removing the referencing here wasn't allowed by the compiler as valid syntax (i.e. neither of the following compiled:
```
    case_likelihood_cache: 'a mut BTreeMap<AlleleFreq, LogProb>,
    case_likelihood_cache: mut BTreeMap<AlleleFreq, LogProb>,
```
And while it could definitely be that I just don't get the syntax right, the `Cell` solution nevertheless seems to make more sense to me.

Reply:
Just this: `case_likelihood_cache: BTreeMap<AlleleFreq, LogProb>,`. And then, `&mut self` in methods that change the cache. As a rule of thumb: interior mutability (Cell) should only be used if it cannot be avoided, because you (a) generate runtime overhead, and (b) constrain potential parallelism we might add in the future.